lib/transaction/pending.go

```diff
@@ -0,0 +1,125 @@
+// Implement operations related to pending operations
+//
+// Pending operations are recorded blockchain operations that
+// have not yet altered the state of the node,
+// but will trigger at a specific block height.
+package transaction
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+)
+
+// The main type is a simple list with insert/pop primitives
+type PendingPool struct {
+	sync.RWMutex
+	*list.List
+}
+
+// Unexported item type
+// We construct it from a height/hash pair,
+// but the user only needs to get the hash back
+// as popping requires the height
+type pendingPoolItem struct {
+	// Block height at which this operation will take action
+	height uint64
+	// Operation hash
+	txHash string
+}
+
+// Instantiate a new `PendingPool`
+func NewPendingPool() *PendingPool {
+	return &PendingPool{sync.RWMutex{}, list.New()}
+}
+
+// Insert a new element in the pool
+func (pp *PendingPool) Insert(height uint64, hash string) {
+	elem := &pendingPoolItem{height: height, txHash: hash}
```

Comment:
The second argument of `Insert()` is `opKey`, but parameter name is `hash` and it will be assigned to txHash.
`opKey` to `txHash`? I'm confused. Is there any reason?
As far as I know, `opKey` is "opHash-txHash"
Therefore, I figure just `opKey` is better.